### PR TITLE
Add unit test to transaction record map

### PR DESCRIPTION
**Pull request by zancas**
This depends on #911 #912 #913 so unless all of those pass review and merge this shouldn't.

Because the others require attention first, I have not yet requested review on this one.

### CODE REVIEW

**Review by fluidvanadium — DISMISSED** [review]
tested and acknowledged!

this is a wicked mocker pattern. i love it

**Comment by fluidvanadium** [comment]
what thingie?

**Comment by zancas** [comment]
Hello, I have resolved conflicts and merged latest dev.

File: zingolib/src/wallet/transaction_record.rs [hunk 1]
```diff
@@ -171,6 +172,43 @@ impl TransactionRecord {
             self.total_orchard_value_spent,
         ]
     }
+    pub fn get_received_note<D>(
```
Comment (fluidvanadium) [inline zingolib/src/wallet/transaction_record.rs]: this fn is extra

**Review by fluidvanadium — DISMISSED** [review]

File: zingolib/src/wallet/transaction_record.rs (hunk 1 above)
Comment (fluidvanadium) [reply zingolib/src/wallet/transaction_record.rs]: i mean, it is superceded by the stuff in #926  and wont be implemented

**Review by fluidvanadium — APPROVED** [review]
this is great! maybe i can duplicate these patterns for some more mocking stuff

**Pull request merged**